nixos/modules/services/networking/keepalived/default.nix

```
@@ -282,16 +295,25 @@ in
       };
     };
 
-    systemd.services.keepalived = {
+    systemd.services.keepalived = let
+      finalConfigFile = if cfg.secretFile == null then keepalivedConf else "/var/run/keepalived/keepalived.conf";
+    in {
       description = "Keepalive Daemon (LVS and VRRP)";
       after = [ "network.target" "network-online.target" "syslog.target" ];
       wants = [ "network-online.target" ];
       serviceConfig = {
         Type = "forking";
         PIDFile = pidFile;
         KillMode = "process";
+        RuntimeDirectory = "keepalived";
+        ExecStartPre = lib.optional (cfg.secretFile != null)
+        (pkgs.writeShellScript "pre-start" ''
```

```suggestion
        (pkgs.writeShellScript "keepalived-pre-start" ''
```

That's a bit more meaningful, when you see it scrolling past.